src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -2305,8 +2306,20 @@ object Desugar {
       }
     }
 
-    def singleAss(left: in.Assignee, right: in.Expr)(info: Source.Parser.Info): in.SingleAss = {
-      in.SingleAss(left, implicitConversion(right.typ, left.op.typ, right))(info)
+    def singleAss(left: in.Assignee, right: in.Expr, isInitExpr: Boolean = false)(info: Source.Parser.Info): in.Stmt = {
+      if (isInitExpr) {
+        val newInfo = info match {
+          case s: Source.Parser.Single => s.createAnnotatedInfo(InhaleInsteadOfAssignmentAnnotation)
+          case i => violation(s"l.op.info ($i) is expected to be a Single")
+        }
+        // Optimization: if we know that right is the expr passed to the declaration of left, then there is no need to
+        // assign to left (which implies exhaling and inhaling the footprint). Instead, we can just assume directly the
+        // equality between left and right.
+        val eq = in.ExprAssertion(in.GhostEqCmp(left.op, implicitConversion(right.typ, left.op.typ, right))(newInfo))(newInfo)
+        in.Inhale(eq)(newInfo)
```

Comment:
If it is ultra urgent, then you could introduce another internal node, e.g. "InitialAssignment" and then replace the inhales with that. That would still be an ugly solution, but at it maintaines the separation of purposes somewhat.